### PR TITLE
fix aria-hidden on anchor

### LINKED ISSUE #7459: Accessibility Issue on Layer Component: Focus Retained on Element with aria-hidden
### Is this something you or your team is interested in contributing a pull request towards?

Yes

### Expected Behavior

Layer component with modal set to true should conform to accessibility guidelines.

### Actual Behavior

The following console error is displayed when opening the modal:

```
Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
Element with focus: a
Ancestor with aria-hidden:  <a tabindex=​"-1" aria-hidden=​"true" class=​"LayerContainer__HiddenAnchor-sc-1srj14c-0 kwDYmk">​</a>​
```

It seems an ancestor of the hidden layer anchor is also set with `aria-hidden=​"true"`, causing the error.

### URL, screen shot, or Codepen exhibiting the issue

Screenshot 2024-12-09 at 23 45 30

### Steps to Reproduce

1. go to https://v2.grommet.io/layer
2. Open dev tools console
3. click the "Show" button. 

### Your Environment

- Grommet version: 2 - whatever's on the component playground at https://v2.grommet.io/layer
- Browser Name and version: Chrome Version 131.0.6778.109 
- Operating System and version (desktop or mobile): Mac OS 14.7.1 desktop

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR fixes an console log error that was happening within Layer 
```
Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
Element with focus: a
Ancestor with aria-hidden:  <a tabindex=​"-1" aria-hidden=​"true" class=​"LayerContainer__HiddenAnchor-sc-1srj14c-0 kwDYmk">​</a>​
```
#### Where should the reviewer start?
LayerContainer.js
#### What testing has been done on this PR?
locally check for console log
#### How should this be manually tested?
storybook 
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
```
 Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus
```
The anchor element is focusable but is being hidden from assistive technologies with `aria-hidden="true"` 

using `inert` will prevent focus and hide the element from assistive technologies which is what we intend. 
#### What are the relevant issues?
closes #7459 
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/DataFilters/__tests__/__snapshots__/DataFilters-test.tsx.snap [hunk 1]
```diff
@@ -1671,8 +1671,8 @@ exports[`DataFilters layer 2`] = `
     id="test-data--filters-layer"
   >
     <a
-      aria-hidden="true"
       class="c3"
+      inert=""
```
Comment (jfeferman) [inline src/js/components/DataFilters/__tests__/__snapshots__/DataFilters-test.tsx.snap:1675]: I believe the markup could be just `<a class="c3" tabindex="-1" inert>`

**Comment by taysea** [comment]
@britt6612 Where the suggestion says "Consider using the inert attribute instead, which will also prevent focus", is it incorrect HTML to be programmatically placing focus on an inert element? When I interact with [this story](https://deploy-preview-7460--sad-tereshkova-173d07.netlify.app/?path=/story/layout-layer-target--target-layer), the focus isn't moved to the layer container, so I can't get into the layer


https://github.com/user-attachments/assets/aab0acb1-a9b4-44f8-9c85-516dc3888147

**Comment by britt6612** [comment]
> Details

@taysea 
Im not seeing this locally I wonder if its related to this comment we saw before?

https://github.com/grommet/grommet/pull/7324#issuecomment-2384321106

**Comment by jfeferman** [comment]
In the [referenced story](https://deploy-preview-7460--sad-tereshkova-173d07.netlify.app/?path=/story/layout-layer-target--target-layer), when the modal opens, the first child `<div>` of `storybook-root` — referred to here as the ancestor — is programmatically set to `aria-hidden="true"`.

However, if you set modal prop to "false", this ancestor is not changed and no error is displayed in the console.  Is it necessary to set `aria-hidden="true"` on the ancestor for the modal option to function correctly?

**Comment by taysea** [comment]
> > Details
> 
> @taysea Im not seeing this locally I wonder if its related to this comment we saw before?
> 
> [#7324 (comment)](https://github.com/grommet/grommet/pull/7324#issuecomment-2384321106)

You're right, it must've been related to that. If I close out the netlify deploy bar prior to interacting, then I don't have the issue.

**Comment by britt6612** [comment]
updated PR:


I removed the `hiddenAnchor` 

This is because the `hiddenAnchor` was initially added for the following reason:
```
      // once layer is open we set the focus in the hidden
      // anchor so that you can start tabbing inside the layer
```

Since this was added we have enhanced our `FocusedContainer.js` to be able to trapFocus within a modal. Instead of the `hiddenAnchor` we know we have 2 `divs` that are pre and post of the modal that is open that is there in order to make sure the `tab` stays within the modal.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by halocline — APPROVED** [review]

**Pull request merged**